packages/engine-server/src/DendronEngineV3.ts

```diff
@@ -647,6 +673,173 @@ export class DendronEngineV3 implements DEngine {
     };
   }
 
+  /**
+   * See {@link DEngine.renameNote}
+   */
+  async renameNote(opts: RenameNoteOpts): Promise<RespV2<RenameNotePayload>> {
+    const ctx = "DEngine:renameNote";
```

Comment:
Talked to Kevin briefly on this before and we'd like this to be atomic as well. Since the current behavior is not atomic, I won't modify it yet, but I'll leave a todo for us to pick up after migration is complete